packages/wallet-connectors/src/WalletConnect.ts

```diff
@@ -38,30 +40,172 @@ import { UnreachableCaseError } from './error';
+} satisfies WalletConnectModalMobileWallet;
+
+const DEFAULT_MOBILE_WALLETS = {
+    [TESTNET.name]: [CONCORDIUM_WALLET_TESTNET, CRYPTO_X_WALLET_TESTNET],
+    [MAINNET.name]: [CONCORDIUM_WALLET_MAINNET, CRYPTO_X_WALLET_MAINNET],
+};
+
+/**
+ * Creates a {@linkcode WalletConnectModalConfig}.
+ *
+ * @param network The {@linkcode Network} to connect to.
+ * @param [mobileWallets] The list of mobile wallets to use for deep linking. Defaults to the concordium and cryptoX wallets for mobile for the specified `network`.
+ * If `network` is anything but {@linkcode MAINNET} or {@linkcode TESTNET}, the default value is undefined.
+ * @param [enableExplorer] Whether to enable the wallet connect explorer in the wallet connect modal. Defaults to `false`.
+ *
+ * @returns the corresponding {@linkcode WalletConnectModalConfig}
+ */
+export function createWalletConnectModalConfig(
+    network: Network,
+    mobileWallets: WalletConnectModalMobileWallet[] = DEFAULT_MOBILE_WALLETS[network.name],
```

Comment:
How does this type check? The default value lookup can fail but the type is not nullable. 🤔

Reply:
Good question, and good catch 😄 

Added a fallback to the fallback 🤔